tests/layer1/test_port_error.py

```diff
@@ -44,23 +45,29 @@ def get_mac_fault_count(dut, interface, fault_type):
     def get_interface_status(dut, interface):
         return dut.show_and_parse("show interfaces status {}".format(interface))[0].get("oper", "unknown")
 
-    @pytest.fixture
+    @pytest.fixture(scope="class")
     def select_random_interfaces(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname):
         dut = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
-        interfaces = list(dut.show_and_parse("show interfaces status"))
 
         sfp_presence = dut.command(cmd_sfp_presence)
         parsed_presence = {line.split()[0]: line.split()[1] for line in sfp_presence["stdout_lines"][2:]}
 
-        available_interfaces = [
-            intf["interface"] for intf in interfaces
-            if parsed_presence.get(intf["interface"]) == "Present"
-        ]
+        eeprom_infos = dut.command("sudo sfputil show eeprom -d")['stdout']
+        eeprom_infos = parse_sfp_eeprom_infos(eeprom_infos)
 
-        pytest_assert(available_interfaces, "No interfaces with SFP detected. Cannot proceed with tests.")
+        available_optical_interfaces = []
```

Comment:
@AntonHryshchuk can we make it attribute driven? Each test user have different testbed and different criteria to pass. For eg, For one user wants to run test_port_error on all 512 ports and some only to selected ports. Some want to run the iteration for 100 times and some may be 10 or 1 . Can we make these variations as attribute input to this test?